src/protocol.h

```diff
@@ -354,28 +354,43 @@ static inline bool MayHaveUsefulAddressDB(ServiceFlags services)
     return (services & NODE_NETWORK) || (services & NODE_NETWORK_LIMITED);
 }
 
+enum class AddrSerialization {
+    DISK,
+
+    // The only time we serialize a CAddress object without nTime is in
+    // the initial VERSION messages which contain two CAddress records.
+    // At that point, the serialization version is INIT_PROTO_VERSION.
+    // After the version handshake, all ADDR messages are serialized with
+    // nTime.
+    NETWORK_NOTIME,
+    NETWORK_WITHTIME
```

Comment:
in the last commit:

Could add a trailing semicolon? I've prepared a suggested diff for you that also switches to a switch-case so that the compiler can check all cases are covered, feel free to take or reject:


```diff
diff --git a/src/protocol.h b/src/protocol.h
index 500cf346e9..0f5786a1a9 100644
--- a/src/protocol.h
+++ b/src/protocol.h
@@ -356,14 +356,8 @@ static inline bool MayHaveUsefulAddressDB(ServiceFlags services)
 
 enum class AddrSerialization {
     DISK,
-
-    // The only time we serialize a CAddress object without nTime is in
-    // the initial VERSION messages which contain two CAddress records.
-    // At that point, the serialization version is INIT_PROTO_VERSION.
-    // After the version handshake, all ADDR messages are serialized with
-    // nTime.
     NETWORK_NOTIME,
-    NETWORK_WITHTIME
+    NETWORK_WITHTIME,
 };
 
 /** A CService with information about it as peer */
@@ -383,13 +377,23 @@ public:
     SERIALIZE_METHODS_PARAMS(CAddress, obj, AddrSerialization, fmt)
     {
         SER_READ(obj, obj.nTime = TIME_INIT);
-        if (fmt == AddrSerialization::DISK) {
+        switch (fmt) {
+        case AddrSerialization::DISK: {
             uint32_t disk_version = DISK_VERSION;
             READWRITE(disk_version);
             READWRITE(obj.nTime);
-        } else if (fmt == AddrSerialization::NETWORK_WITHTIME) {
+            break;
+        }
+        case AddrSerialization::NETWORK_WITHTIME: {
             READWRITE(obj.nTime);
+            break;
+        }
+        case AddrSerialization::NETWORK_NOTIME: {
+            // The only time we serialize a CAddress object without nTime is in
+            // the initial VERSION messages which contain two CAddress records.
+            break;
         }
+        } // no default case, so the compiler can warn about missing cases
         READWRITE(Using<CustomUintFormatter<8>>(obj.nServices), AsBase<CService>(obj));
     }

Reply:
Included.